### DISCUSSION TITLE
feature request: Prevent `clear` command from clearing scrollback history

### DISCUSSION THREAD

**Discussion by rrotter** [ghostty-org/ghostty#3927] [Feature Requests, Ideas] [answered]
There is a setting in iTerm to "Prevent CSI 3 J from clearing scrollback history", aka `defaults write com.googlecode.iterm2 PreventEscapeSequenceFromClearingHistory true`. When set, the `clear` command will scroll the current prompt to the top of the screen, but the scrollback buffer stays intact. This is really useful when I ssh into poorly configured servers (i.e. root's `.bash_logout` contains `clear`) and want to prevent them from vandalizing my terminal, or I just want to have the option to clear the screen without clearing scrollback. I use both `clear` and `⌘K` depending upon whan I want to do, and I'd like these both to work.

**Comment by chenzhiwei** [comment, marked as answer]
Related: https://github.com/ghostty-org/ghostty/discussions/5269

**Reply by rhodes-b** [reply]
`ctrl+l` will preserve scrollback, many shells support this

**Reply by rrotter** [reply]
Yes, it does, and I use it. But, it doesn't stop a program running in the terminal from clearing the history. That's control I'd rather not give to the remote host.

I guess, more broadly what I'm asking for is:
"I see no valid purpose for `clear`, so please give me the option to always treat it the same as `^L`"

**Reply by chenzhiwei** [reply]
ctrl+l in local shell will preserve scrollback and the current screen; in ssh shell only preserves the scrollback.

**Comment by taylrfnt** [comment]
See #4929

**Reply by rrotter** [reply]
#4929 is a question about whether the current behavior is a bug. It's not a bug, fine. 

This thread is a request to add an option to change the current behavior.
